src/main/kotlin/com/aemtools/completion/util/HtlUtil.kt

```diff
@@ -31,4 +34,47 @@ fun extractItemAndItemListNames(value: String): Pair<String, String> {
         }
     }
     return item to "${item}List"
+}
+
+/**
+ * Check if current string literal is the main EL's string
+ * (e.g. ${'main string' @ param='not main string'})
+ */
+fun HtlStringLiteral.isMainString(): Boolean {
+    val expression = this.findParentByType(com.aemtools.lang.htl.psi.HtlExpression::class.java) ?: return false
+
+    return expression.parent is HtlHel
+}
+
+/**
+ * Check if current variable is "option"
+ */
+fun HtlVariableName.isOption() : Boolean {
+    return this.hasParent(HtlContextExpression::class.java)
+        && !this.hasParent(HtlAssignment::class.java)
+}
+
+/**
+ * Extract first (top level) [PropertyAccessMixin].
+ */
+fun HtlHtlEl.extractPropertyAccess(): PropertyAccessMixin? {
```

Comment:
The prefix is declared in "Htl.bnf" file:

```
psiClassPrefix="Htl"
```

The name of corresponding rule is htl_el (`htl_el ::= ( hel_start hel? context? hel_end?) {recoverWhile="recover_rule" pin=1}`)

So, we have `HtlHtlEl` after generation.

Don't see the way how the element could be renamed